src/onnx/onnx.cpp

```diff
@@ -2486,6 +2488,28 @@ struct onnx_parser
         return l;
     }
 
+    instruction_ref
+    parse_greater(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        auto l = add_broadcastable_binary_op(args[0], args[1], "greater");
+        if(l->get_shape().type() != shape::bool_type)
+        {
+            l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
+        }
+        return l;
+    }
+
+    instruction_ref
+    parse_less(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        auto l = add_broadcastable_binary_op(args[0], args[1], "less");
+        if(l->get_shape().type() != shape::bool_type)
+        {
+            l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
+        }
+        return l;
+    }
+
```

Comment:
the combined function would be like:

instruction_ref parse_compare_op(const std::string&,
                             const std::string& op_name,
                             const node_info&,
                             std::vector<instruction_ref> args)
{
    auto l = add_broadcastable_binary_op(args[0], args[1], op_name);
    if(l->get_shape().type() != shape::bool_type)
    {
        l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
    }
    return l;
}

Then the add_mem_op is called like:

add_mem_op("Equal", "equal", &onnx_parser::parse_compare_op);
add_mem_op("Greater", "greater", &onnx_parser::parse_compare_op);
add_mem_op("Less", "less", &onnx_parser::parse_compare_op);

Reply:
this is also the main suggested code change I had.

Reply:
Added